XenAdmin/Wizards/NewVMWizard/Page_Storage.cs

```diff
@@ -167,88 +170,79 @@ private void LoadDisks()
                         SR = new XenRef<SR>(sr != null ? sr.opaque_ref : Helper.NullOpaqueRef)
                     };
 
-                    rowList.Add(new DiskGridRowItem(Connection, disk, device, DiskSource.FromCustomTemplate));
+                    var row = new DiskGridRowItem(Connection, disk, device, DiskSource.FromCustomTemplate);
+                    row.UpdateStatus(icon, tooltip);
+                    rowList.Add(row);
                 }
             }
 
             DisksGridView.Rows.AddRange(rowList.ToArray());
+            UpdateStatusForEachDisk(true);
         }
 
         /// <summary>
         /// Tries to find the best SR for the given VDI considering first the
         /// suggestedSR then the pool's default SR, then other SRs.
         /// </summary>
         /// <returns>The SR if a suitable one is found, otherwise null</returns>
-        private static SR GetBestDiskStorage(IXenConnection connection, long diskSize, Host affinity, SR suggestedSR)
+        private SR GetBestDiskStorage(IXenConnection connection, long diskSize, Host affinity, SR suggestedSR,
+            out Image icon, out string tooltip)
         {
+            icon = Images.StaticImages._000_VirtualStorage_h32bit_16;
+            tooltip = null;
+
             if (suggestedSR != null && suggestedSR.CanBeSeenFrom(affinity) &&
                 suggestedSR.VdiCreationCanProceed(diskSize))
                 return suggestedSR;
 
             SR defaultSR = connection.Resolve(Helpers.GetPoolOfOne(connection).default_SR);
             if (defaultSR != null && defaultSR.CanBeSeenFrom(affinity) &&
                 defaultSR.VdiCreationCanProceed(diskSize))
+            {
+                if (suggestedSR != null)
+                {
+                    tooltip = string.Format(Messages.NEWVMWIZARD_STORAGEPAGE_SUGGESTED_NOSPACE, suggestedSR.Name().Ellipsise(50));
+                    icon = Images.StaticImages._000_Alert2_h32bit_16;
+                }
                 return defaultSR;
+            }
 
             foreach (SR sr in connection.Cache.SRs)
             {
                 if (sr.CanCreateVmOn() &&
                     sr.CanBeSeenFrom(affinity) && sr.VdiCreationCanProceed(diskSize))
+                {
+                    if (suggestedSR != null && defaultSR != null)
+                        tooltip = Messages.NEWVMWIZARD_STORAGEPAGE_NOSPACE;

```

Comment:
When the suggestedSR is the default SR, the more suitable message would be NEWVMWIZARD_STORAGEPAGE_SUGGESTED_NOSPACE.

Reply:
I realised there is one more problem in the logic, namely when the VM has no home server.